player/src/main/java/com/focusit/jsflight/player/ui/MainFrame.java

```diff
@@ -1071,12 +1061,25 @@ private void initUIFromOptionsController()
         proxyPort.setText(OptionsController.getInstance().getProxyPort());
         ffPath.setText(OptionsController.getInstance().getFfPath());
         pjsPath.setText(OptionsController.getInstance().getPjsPath());
-        maxStepDelayField.setText(OptionsController.getInstance().getMaxStepDelay());
+        pageReadyTimeoutField.setText(OptionsController.getInstance().getPageReadyTimeout());
         makeShots.setSelected(OptionsController.getInstance().getMakeShots() != null
                 ? OptionsController.getInstance().getMakeShots().equalsIgnoreCase("true") : false);
         screenDirTextField.setText(OptionsController.getInstance().getScreenDir());
         checkPageJs.setText(OptionsController.getInstance().getCheckPageJs());
         webDriverTag.setText(OptionsController.getInstance().getWebDriverTag());
+
+        boolean useFF = Boolean.valueOf(OptionsController.getInstance().isUseFirefox());
+        boolean usePJ = Boolean.valueOf(OptionsController.getInstance().isUsePhantomJs());
+        if (!useFF && !usePJ)
```

Comment:
IMAO it is too complicated.
just us FF by default in any case. and then check cli options and use a browser according to specified option. 
If user had chosen both - he had his only trouble